### ISSUE TITLE
manylinux support is missing for riscv64

### ISSUE THREAD

**Issue by markdryan** [astral-sh/uv#8889]
* The current uv platform: linux/riscv64
* The current uv version (`uv --version`): uv 0.4.30 (which includes #8660)

It's not currently possible to install riscv64 manylinux wheels using uv.  If I try locally with the command

```
$ uv pip install PyYAML-6.0.2-cp311-cp311-manylinux_2_35_riscv64.whl 
Using Python 3.11.4 environment at /home/HOME/uv-venv
Resolved 1 package in 156ms
error: Failed to determine installation plan
  Caused by: A path dependency is incompatible with the current platform: PyYAML-6.0.2-cp311-cp311-manylinux_2_35_riscv64.whl
```

I get the error shown.   Although the manylinux container images do not yet support riscv64, auditwheel [does](https://github.com/pypa/auditwheel/blob/main/src/auditwheel/policy/manylinux-policy.json#L446), so it is possible to create manylinux wheels for riscv64.  These wheels can be installed by pip, but not by uv.

According to the auditwheel policy file for manylinux, the oldest version of manylinux that supports riscv64 is [2.31](https://github.com/pypa/auditwheel/blob/main/src/auditwheel/policy/manylinux-policy.json#L326).  Locally, I have tried modifying [get_minimum_manylinux_minor](https://github.com/astral-sh/uv/blob/main/crates/uv-platform-tags/src/platform.rs#L111C12-L111C39) so that it returns 31 for riscv64.  Doing so allows the command shown above to work correctly, e.g.,

```
$ uv pip install PyYAML-6.0.2-cp311-cp311-manylinux_2_35_riscv64.whl 
Using Python 3.11.4 environment at /home/HOME/uv-venv
Resolved 1 package in 165ms
Prepared 1 package in 174ms
Uninstalled 1 package in 14ms
Installed 1 package in 37ms
 - pyyaml==6.0.2
 + pyyaml==6.0.2 (from file:///home/HOME/PyYAML-6.0.2-cp311-cp311-manylinux_2_35_riscv64.whl)
```

I'd like to submit this small change to uv to enable it to install manylinux wheels on riscv64.  Would such a patch be acceptable?

Note, that [RISE](https://riseproject.dev/) is currently building and distributing a small set of riscv64 binary wheels on its [gitlab](https://gitlab.com/riseproject/python/wheel_builder).    The majority of these wheels are manylinux_2_35 wheels and they cannot currently be installed by uv.  They can however, be installed using pip >=24.1.

**Comment by konstin** [comment]
A PR to add risc64 manylinux support would be great!

It's totally not a requirement for the PR, but if you happen to have instructions how to test riscv things on an x86_64 or aarch64 machine, this helps the team to reproduce and debug riscv-specific problems (i've for example used qemu docker for other architectures before).

**Comment by markdryan** [comment]
> 's totally not a requirement for the PR, but if you happen to have instructions how to test riscv things on an x86_64 or aarch64 machine, this helps the team to reproduce and debug riscv-specific problems (i've for example used qemu docker for other architectures before).

I used a combination of a riscv64 container image running on an amd64 host to build the uv wheel which I then tested on a riscv64 VM running Ubuntu 23.10.   The riscv64 wheels distributed by RISE are also built and tested inside riscv64 container images running on an amd64 host.

A few observations.  I did try and run the uv tests in the riscv64 container image but I got a fair few failures (with and without the manylinux patch).  Some of these were timeouts and some seemed to be caused by missing riscv64 packages at PyPi.  The build of the wheel itself probably took about an hour with maturin.

I did try originally building and testing on a VisionFive2 (a riscv64 SBC) running Ubuntu 24.04 but the build was killed after about an hour due to an OOM error.  My VisionFive2 only has 8GB of RAM.
